File: app/src/main/java/com/example/retrofitdemoapp/UserDetailsActivity.java
package com.example.retrofitdemoapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

public class UserDetailsActivity extends AppCompatActivity {

    TextView username, firstname, lastname, email;
    UserResponse userResponse;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        username = (TextView)findViewById(R.id.uName);
        firstname = (TextView)findViewById(R.id.firstName);
        lastname = (TextView)findViewById(R.id.lastname);
        email = (TextView)findViewById(R.id.email);

        Intent intent = getIntent();
        if(intent.getExtras() != null) {
            userResponse = (UserResponse) intent.getSerializableExtra("data");
            String usernameData = userResponse.getUsername();
            int firstnameData = userResponse.getId();
            String lastnameData = userResponse.getDate_joined();
            String emailData = userResponse.getUrl();

            username.setText(usernameData);
            firstname.setText(firstnameData+"");
            firstname.setTextColor(this.getResources().getColor(R.color.colorPrimaryDark));
            lastname.setText(lastnameData);
            lastname.setTextColor(this.getResources().getColor(R.color.colorPrimaryDark));
            email.setText(emailData);
            email.setTextColor(this.getResources().getColor(R.color.colorPrimaryDark));

        }
    }
}

File: app/src/main/java/com/example/retrofitdemoapp/UserDetailsActivity.kt
package com.example.retrofitdemoapp

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.fragment_user_details.*

class UserDetailsActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if (intent.extras?.containsKey("data") == true) {
            val userResponse = intent.getSerializableExtra("data") as UserResponse
            val usernameData = userResponse.username
            val firstnameData = userResponse.id
            val lastnameData = userResponse.date_joined
            val emailData = userResponse.url

            uName.text = usernameData
            firstName.text = firstnameData.toString()
            firstName.setTextColor(resources.getColor(R.color.colorPrimaryDark))
            lastname.text = lastnameData
            lastname.setTextColor(resources.getColor(R.color.colorPrimaryDark))
            email.text = emailData
            email.setTextColor(resources.getColor(R.color.colorPrimaryDark))
        }
    }
}
